feat(analyzer): add a conditional return type for json_encode()

mytskine opened the pull request:
## 📌 What Does This PR Do?

The default return type of PHP's json_encode() is string|false. But when the function receives the flag JSON_THROW_ON_ERROR, then the return type becomes string. So a new handler was created for ext/json, with a special case for the function json_encode().

## 🔍 Context & Motivation

Closes #425

## 🛠️ Summary of Changes

- **Bug Fix:** Fixed incorrect return type of PHP `json_encode(..., JSON_THROW_ON_ERROR)`.

## 📂 Affected Areas

- [x] Analyzer

## 🔗 Related Issues or PRs

#425

azjezz commented on crates/analyzer/src/invocation/special_function_like_handler/standard/json.rs:
```diff
@@ -0,0 +1,56 @@
+                        true,
+                        false,
+                    ))))
+                } else {
```
The resulting type should be a generic `string`, not a `literal-string`.

I think `non-empty-string` works too, AFAIK, all values ( even "empty" ones ) will produce a non-empty string.

mytskine commented:
I replaced both call to `TString::unspecified_literal_with_props()` with `TString::new()` so that the string is just a non-empty string. Then I rebased on the latest main branch.
